### ISSUE TITLE
Automatically use mypy.ini or pyproject.toml when not in the root directory of the workspace

### ISSUE THREAD

**Issue by theomessin** [microsoft/vscode-mypy#190]
When using the Mypy linter, our standard is to have a `mypy.ini` in the project's root with a bunch of config values. Currently we have our developers use this setting for vscode:
```json
  "python.linting.mypyArgs": [
    "--config-file=${workspaceFolder}/mypy.ini"
  ]
```

_However, if a project doesn't have a `mypy.ini` file linting fails silently (which can cause a lot of issues as a dev might think that mypy worked fine)._

I think that auto-detecting and using the `mypy.ini` file should be a part of the plugin itself.

**Comment by karthiknadig** [comment]
There is a `cwd` setting that can be used to set the this. By default, it is the workspace directory, if you want it to run in the same directory as the file you can set it to `${fileDirname}`, and cwd will change to the parent directory of the file.

**Comment by evbo** [comment]
@karthiknadig can we please reopen? `${fileDirname}` is not working for very many cases, such as when you have multiple sub folders.

What does seem to work without any hiccups is the following:

```json

    "mypy-type-checker.args": [
        "--config-file=${workspaceFolder}/pyproject.toml"
    ]
```

However, it would be nice not to have to specify this since it is quite boilerplate configuration

**Comment by karthiknadig** [comment]
@evbo Can you elaborate on your scenario? like what is the directory structure and where are your config files.

From your what is see above it looks like you have a single `pyproject.toml` in your workspace folder. that means you don't need to specify `mypy-type-checker.args`. The `cwd` by default is your workspace folder, and `mypy` should pickup `pyproject.toml`. 

The `${fileDirname}` is useful if you have different configurations for different parts of your project. From what I understand that is not your case.

**Comment by evbo** [comment]
It is a mono repo, like:

```shell
project_root
    python_module_a
        pyproject.toml
        venv
        folder
            source.py
        source.py
    python_module_b
```

**Comment by karthiknadig** [comment]
Do you have a settings file you can share? and also do you have or tried using `*.code-workspace`?

**Comment by karthiknadig** [comment]
@evbo This really is a bug on `mypy` to add the ability to find config in parent directories (similar to pylint, ruff, flake8, etc). I can see that this might be useful until such support is added to `mypy`.

I recommend creating an issue, and we are open to a PR on this.
Things to consider:
1. check the args and if it does not already have a config, and the cwd does not have a config initiate the search.
2. Ensure that this search happens every time for `mypy` case, but produces and warning for `dmypy`. `dmypy` does not re-read configuration.

The changes should happen in here: 
https://github.com/microsoft/vscode-mypy/blob/85a7af33526bca54e327e0b451f1922050cba3a0/bundled/tool/lsp_server.py#L204

**Comment by alanwilter** [comment]
I have:
```json
  "mypy-type-checker.args": [
    "--config-file=${workspaceFolder}/pyproject.toml"
  ],
```

And in my `pyproject.toml`:
```toml
[tool.mypy]
python_version = "3.10"
exclude = 'tests/'
strict = true
disable_error_code = ["type-arg"]
```

If I run on terminal:
```shell
$ mypy test
There are no .py[i] files in directory 'tests'
```
It does ignore `tests`, however, `ms-python.mypy-type-checker` doesn't:

<img width="652" alt="Image" src="https://github.com/user-attachments/assets/2594dd7b-2aca-4425-aad7-af1f906a8a7c" />

But the funny thing is, if I do:
```toml
[tool.mypy]
python_version = "3.10"
# exclude = 'tests/'
strict = true
disable_error_code = ["type-arg", "no-untyped-def"]
```
Presto!

<img width="554" alt="Image" src="https://github.com/user-attachments/assets/d8277e20-e6f5-4a4b-a07e-758e54b1553a" />

So, what's going on here?

**Comment by karthiknadig** [comment]
@alanwilter This happens because by default the extension runs `mypy` on individual files. If you see the logs it would have run this with a cli like this `mypy --config-file=<project-path>/pyproject.toml <file-path>`.

If you set the `mypy-type-checker.reportingScope` to `workspace` you might see different results. With `workspace` scope, the cli will look like `mypy --config-file=<project-path>/pyproject.toml <project-path>`.

Exclusions are not considered by `mypy` when you run it on a file explicitly.

**Comment by alanwilter** [comment]
Thanks @karthiknadig, that solved my issue.
